src/tsutil/Regex.cc

```diff
@@ -105,12 +120,20 @@ class RegexContext
 thread_local RegexContext *RegexContext::_regex_context = nullptr;
 
 //----------------------------------------------------------------------------
-namespace
+
+RegexContextCleanup::~RegexContextCleanup()
 {
```

Comment:
I think you need to acquire `_mutex` here, since it's modifying `_contexts`, right?

Reply:
The mutex is not really needed, since it is called atexit, but I added it anyways.